### ISSUE TITLE
[Bug]: Attempts to collide or enter even though the path is preempted.

### ISSUE THREAD

**Issue by gmchoi96** [open-rmf/rmf#541]
### Before proceeding, is there an existing issue or discussion for this?

- [X] I have done a search for similar [issues](.?q=type%3Aissue) and [discussions](https://github.com/open-rmf/rmf/discussions).

### OS and version

ubuntu22.04

### Open-RMF installation type

Binaries

### Open-RMF version or commit hash

2.3.0

### ROS distribution

Humble

### ROS installation type

Source build

### Description of the bug

I am running a simulation in which 4 TinyRobots repeat pick and drop actions at specific points.. However, as seen in the video, there are symptoms of a collision or an attempt to enter even though the path has been preempted.

[collision.webm](https://github.com/user-attachments/assets/3a0254d8-f376-4465-8589-ecb4f14665e6)
[enter.webm](https://github.com/user-attachments/assets/922ceb3e-e556-4db9-a410-944221d86b36)

### Steps to reproduce the bug

1. Run TinyRobot simulation

**Comment by mxgrey** [comment]
Thanks for the report. Are you able to upload the repo with the map and any launch file you used, that way we can debug the scenario directly?

**Comment by gmchoi96** [comment]
Hello @mxgrey ! Thank you for your answer.

Before more complex tests, I wanted to check the stability of rmf driving, so I modified the demos file provided a little and tested it as shown below.

I will give you the github address so that you can download the file.
https://github.com/gmchoi96/rmf_sort_test/tree/main

The execution command I tested was
[ simulation ]
ros2 launch rmf_demos_gz office.launch.xml server_uri:="ws://localhost:8000/_internal"
[ tasks ] 
ros2 run rmf_demos_tasks dispatch_loop -s tinyRobot1_spawn -f drop13 -n 7 --use_sim_time
ros2 run rmf_demos_tasks dispatch_loop -s tinyRobot2_spawn -f drop9 -n 7 --use_sim_time
ros2 run rmf_demos_tasks dispatch_loop -s tinyRobot3_spawn -f drop6 -n 7 --use_sim_time
ros2 run rmf_demos_tasks dispatch_loop -s tinyRobot4_spawn -f drop3 -n 7 --use_sim_time

I also briefly checked and tested for about 10 minutes, and confirmed that the symptoms appeared, and reported it because the symptoms seemed very dangerous. In addition to this, there were some problems, such as the robot not moving even when a new path was created while driving. I couldn't confirm whether this is a problem that always occurs. However, if it is an intermittent problem, I think it can cause bigger problems during actual testing.

Is this an issue that can be improved? I plan to test it with more complex robots, 10+.

If you have any additional questions, please let me know.

-- Basically, I modified the files provided by rmf, and I will organize the list of files I modified. Check it out if necessary. -- 

[ add ] 
rmf_demos_maps/maps/sorting_test
[ modify ]
rmf_demos/launch/office.launch.xml/
rmf_demos_gz/launch/office.launch.xml
rmf_demos_assets/models/TinyRobot/model.sdf
rmf_demos_fleet_adapter/config.yaml
rmf_demos/config/office/tinyRobot_config.yaml
